src/sentry/tasks/delete_seer_grouping_records.py

```diff
@@ -83,7 +86,9 @@ def call_seer_delete_project_grouping_records(
     *args: Any,
     **kwargs: Any,
 ) -> None:
-    if killswitch_enabled(project_id):
+    if killswitch_enabled(project_id) or options.get(
+        "seer.similarity-embeddings-delete-by-hash-killswitch.enabled"
+    ):
```

Comment:
Any reason we couldn't add the second killswitch check to `killswitch_enabled`? IMHO it would be cleaner if we can - you'd only need to add it in one place.

Reply:
Also, the purist in me says the killswitch bit should technically be in a separate PR, but I don't think it's a huge deal.

Reply:
the `killswitch_enabled` function is used for the seer grouping changes, whereas this separate killswitch check if just for the deletion - do you think it'd make sense to add it to the function and run it based on an input parameter?

Reply:
Yeah, my instinct would be to be able to pass in a project id or feature name or both and have it check whatever killswitches it needs to.

Reply:
If we're going to have feature-specific (which is maybe just endpoint-specific?) killswitches, what I meant was more something like this:

```python
def killswitch_enabled(
    project_id: int, event: Event | None = None, feature: str | None = None
) -> bool:

	# <global and project killswitch checks>

	killswitches_by_feature = {
	    "delete-seer-records": "seer.similarity-embeddings-delete-by-hash-killswitch.enabled"
	}

	if feature and feature in killswitches_by_feature and options.get(killswitches_by_feature[feature]):
		# <logging and/or metrics>
		return True

	# <whatever other checks we might add>

	return False
```

That way, the signature won't change every time we add a new killswitch.

Reply:
But, if you do the above, at that point I actually do think that the killswitch stuff should be in a separate PR, because it's not really about which feature flag to use.